File: javadp - banking/factory/AccountFactory.java
package factory;

import model.*;

public class AccountFactory {

    public static AbstractAccount createAccount(AccountType type, String userId) {

        switch(type) {
            case SAVINGS:
                return new SavingsAccount(userId);
            case CURRENT:
                return new CurrentAccount(userId);
            default:
                throw new IllegalArgumentException("Invalid account type");
        }
    }
}

File: pythondp - banking/factory/account_factory.py
from model.savings_account import SavingsAccount
from model.current_account import CurrentAccount


class AccountFactory:

    @staticmethod
    def create_account(account_type, user_id):

        if account_type == "SAVINGS":
            return SavingsAccount(user_id)

        elif account_type == "CURRENT":
            return CurrentAccount(user_id)

        else:
            raise Exception("Invalid account type")